tests/rule_based_profiler/config/test_base.py

```diff
@@ -279,3 +283,117 @@ def test_rule_based_profiler_from_commented_map():
     commented_map = CommentedMap(data)
     config = RuleBasedProfilerConfig.from_commented_map(commented_map)
     assert all(hasattr(config, k) for k in data)
+
+
+def test_resolve_config_using_acceptable_arguments(
+    profiler_with_placeholder_args: RuleBasedProfiler,
+) -> None:
+    config: RuleBasedProfilerConfig = (
+        RuleBasedProfilerConfig.resolve_config_using_acceptable_arguments(
+            profiler=profiler_with_placeholder_args
+        )
+    )
+
+    old_config: RuleBasedProfilerConfig = profiler_with_placeholder_args.config
+
+    # Brand new config is created but existing attributes are unchanged
+    assert id(old_config) != id(config)
+    assert all(
+        old_config[attr] == config[attr]
+        for attr in ("class_name", "config_version", "module_name", "name")
+    )
+
+
+def test_resolve_config_using_acceptable_arguments_with_runtime_overrides(
+    profiler_with_placeholder_args: RuleBasedProfiler,
+) -> None:
+    rule_name: str = "my_new_rule"
```

Comment:
Non blocking nitpick, up to you - I would name this something more obvious like `runtime_override_rule_name` and rename `rule` to `runtime_override_rule`, `rules` to `runtime_override_rules`, etc.

Reply:
done!